lib/ingram_micro/elements/sales_order_submission.rb

```diff
@@ -1,67 +1,68 @@
-    @element[:sales_order_shipment_information] ||= IngramMicro::SalesOrderShipmentInformation.new
-    @element[:credit_card_information] ||= IngramMicro::CreditCardInformation.new
-    @element[:sales_order_header] ||= IngramMicro::SalesOrderHeader.new
-    check_line_items
-    @element[:detail] ||= IngramMicro::Detail.new({line_items: @element[:line_items]})
-  end
+    def initialize(options = {})
+      super
+      @element[:customer] ||= Customer.new
+      @element[:sales_order_shipment_information] ||= SalesOrderShipmentInformation.new
+      @element[:credit_card_information] ||= CreditCardInformation.new
+      @element[:sales_order_header] ||= SalesOrderHeader.new
+      check_line_items
+      @element[:detail] ||= Detail.new({line_items: @element[:line_items]})
+    end
 
-  def build(builder)
-    builder.send('header') do
-      builder.send 'customer-id', @element[:customer_id]
-      builder.send 'business-name', @element[:business_name]
-      builder.send 'carrier-name', @element[:carrier_name]
-      builder.send('customer-information') do
-        @element[:customer].build(builder)
-      end
-      builder.send('shipment-information') do
-        @element[:sales_order_shipment_information].build(builder)
-      end
-      if @element[:purchase_order_information]
-        builder.send('purchase-order-information') do
-          @element[:purchase_order_information].build(builder)
+    def build(builder)
+      builder.send('header') do
+        builder.send 'customer-id', IngramMicro.configuration.customer_id
+        builder.send 'business-name', @element[:business_name]
+        builder.send 'carrier-name', @element[:carrier_name]
+        builder.send('customer-information') do
+          @element[:customer].build(builder)
+        end
+        builder.send('shipment-information') do
+          @element[:sales_order_shipment_information].build(builder)
+        end
+        if @element[:purchase_order_information]
+          builder.send('purchase-order-information') do
+            @element[:purchase_order_information].build(builder)
+          end
+        end
+        builder.send('credit-card-information') do
+          @element[:credit_card_information].build(builder)
+        end
+        builder.send('order-header') do
+          @element[:sales_order_header].build(builder)
         end
       end
-      builder.send('credit-card-information') do
-        @element[:credit_card_information].build(builder)
-      end
-      builder.send('order-header') do
-        @element[:sales_order_header].build(builder)
+      builder.send('detail') do
+        @element[:detail].build(builder)
       end
     end
-    builder.send('detail') do
-      @element[:detail].build(builder)
-    end
-  end
 
-  def check_line_items
-    if @element[:line_items].empty?
-      line_item = IngramMicro::SalesOrderLineItem.new
-      @element[:line_items] << line_item
+    def check_line_items
```

Comment:
Removing this altogether so that if no line items are passed in, the xml won't pass the schema.